modules/@apostrophecms/rich-text-widget/ui/apos/tiptap-extensions/Div.js

```diff
@@ -0,0 +1,44 @@
+import { mergeAttributes, Node } from '@tiptap/core';
+
+export default (options) => {
+  return Node.create({
+
+    name: 'div',
+
+    addOptions() {
+      return {
+        HTMLAttributes: {}
+      };
+    },
+
+    content: 'block+',
+
+    group: 'block',
+
+    defining: true,
+
+    parseHTML() {
+      return [
+        { tag: 'div' }
+      ];
+    },
+
+    renderHTML({ HTMLAttributes }) {
+      return [ 'div', mergeAttributes(this.options.HTMLAttributes, HTMLAttributes), 0 ];
+    },
+
+    addCommands() {
+      return {
+        setDiv: () => ({ commands }) => {
+          return commands.wrapIn(this.name);
+        },
+        toggleDiv: () => ({ commands }) => {
+          return commands.toggleWrap(this.name);
+        },
+        unsetDiv: () => ({ commands }) => {
+          return commands.lift(this.name);
+        }
+      };
+    }
```

Comment:
This actually lead me to figure out wrapping was the wrong behavior and now my implementation is cribbed from the heading extension because that's what it's really meant to be like. I tried using your setNode feature but I still couldn't get it to toggle from paragraphs rather than wrapping them that way. Works well this way.